mobile/cmp/navigator/NavigatorModel.ts

```diff
@@ -131,8 +140,50 @@ export class NavigatorModel extends HoistModel {
     //--------------------
     // Implementation
     //--------------------
-    private onRouteChange(init = null) {
-        if (!this._navigator || !XH.routerState) return;
+    /** @internal */
+    setSwiper(swiper: Swiper) {
+        if (this._swiper) return;
+        this._swiper = swiper;
+
```

Comment:
think you can replace the `this.swiper` safely  with swiper in the rest of this method  post line 146